src/fleet/azext_fleet/custom.py

```diff
@@ -212,6 +216,89 @@ def get_credentials(cmd,  # pylint: disable=unused-argument
+                 name,
+                 system_assigned=None,
+                 user_assigned=None,
+                 no_wait=False):
+    fleet_patch_model = cmd.get_models(
+        "FleetPatch",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleets"
+    )
+    fleet_managed_service_identity_model = cmd.get_models(
+        "ManagedServiceIdentity",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleets"
+    )
+
+    # Handle all 6 permutations of mi_system & mi_user assigned parameters
+    if system_assigned is True and user_assigned is not None:
+        raise CLIError("A fleet can either have a system or user assigned identity, not both.")
+    if system_assigned is None and user_assigned is None:
+        managed_service_identity = None
+    elif system_assigned is False and user_assigned is None:
+        managed_service_identity = fleet_managed_service_identity_model(type="None")
+    elif system_assigned is True and user_assigned is None:
+        managed_service_identity = fleet_managed_service_identity_model(type="SystemAssigned")
+    else:
+        managed_service_identity = fleet_managed_service_identity_model(type="UserAssigned")
+        user_assigned_identity_model = cmd.get_models(
+                "UserAssignedIdentity",
+                resource_type=CUSTOM_MGMT_FLEET,
+                operation_group="fleets"
+            )
+        managed_service_identity.user_assigned_identities = {user_assigned: user_assigned_identity_model()}
+
+    fleet_patch = fleet_patch_model(
+        identity=managed_service_identity
+    )
+
+    return sdk_no_wait(no_wait, client.begin_update, resource_group_name, name, fleet_patch, polling_interval=5)
+
+
+def remove_fleet_identity(cmd,
+                 client,
+                 resource_group_name,
+                 name,
+                 system_assigned=False,
+                 user_assigned=False,
+                 no_wait=False):
+    fleet_patch_model = cmd.get_models(
+        "FleetPatch",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleets"
+    )
+    fleet_managed_service_identity_model = cmd.get_models(
+        "ManagedServiceIdentity",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleets"
+    )
+
+    if system_assigned is False and user_assigned is False:
+        raise CLIError("Either --system-assigned is set to true, or pass an empty --user-assigned")
+    else:
+        managed_service_identity = fleet_managed_service_identity_model(type="None")
+
+    fleet_patch = fleet_patch_model(
+        identity=managed_service_identity
+    )
+
+    return sdk_no_wait(no_wait, client.begin_update, resource_group_name, name, fleet_patch, polling_interval=5)
+
+
+def show_fleet_identity(cmd,  # pylint: disable=unused-argument
+               client,
+               resource_group_name,
+               name):
+    
+    identity = getattr(client.get(resource_group_name, name), 'identity', [])
```

Comment:
should be `{}`? or is `[]` correct?

Reply:
Yup {}, not an array. The guidelines brought to attention that resources can have multiple identities, which we obviously don't support ATM.